Assets/Scripts/DialogueManager.cs

```diff
@@ -219,72 +261,90 @@ private IEnumerator StartTypeText(string text)
             textMesh.text = text.Substring(0, currentCharIndex);
             yield return new WaitForSeconds(charTypeSpeed);
         }
-
-        waitingForNext = true;
-        while (waitingForNext)
+        activeConversations[conversation].waitingForNext = true;
+        while (activeConversations[conversation].waitingForNext)
         {
             yield return new WaitForFixedUpdate();
         }
 
         textMesh.text = string.Empty;
 
-        skipRequest = false;
-        isTyping = false;
+        activeConversations[conversation].skipRequest = false;
+        activeConversations[conversation].isTyping = false;
+
+        AdvanceConversation(conversation);
+    }
+
+    private void FinishConversation(Conversation conversation)
+    {
+        HideAllSpeakers(conversation);
+
+        // Stop any coroutines that are part of the conversation
+        // Null checks are done to be super safe just incase coroutines have not been set
+        if (activeConversations[conversation].coroutine != null)
+        {
+            StopCoroutine(activeConversations[conversation].coroutine);
+        }
+        if (activeConversations[conversation].typingCoroutine != null)
+        {
+            StopCoroutine(activeConversations[conversation].typingCoroutine);
+        }
+
+        activeConversations.Remove(conversation);
+
+        if (advancingConversation == conversation)
+        {
+            EndAdvancing();
+        }
+    }
+
+    void HideAllSpeakers(Conversation conversation)
+    {
+        List<string> convoSpeakers = conversation.GetAllSpeakers();
 
-        AdvanceConversation();
+        foreach(string speaker in convoSpeakers)
+        {
+            SpeakerUI speakerUI = Utils.GetRequiredComponent<SpeakerUI>(speakers[speaker].speakerObject);
+            speakerUI.Hide();
+        }
     }
 
-    private void PlayVoice(string fmodPath)
+    private void PlayVoice(string fmodPath, SpeakerContainer currentSpeaker)
     {
         if (!string.IsNullOrEmpty(fmodPath.Trim()))
         {
             FMODUnity.RuntimeManager.PlayOneShot(fmodPath, currentSpeaker.speakerObject.transform.position);
         }
     }
 
-    private void TriggerAutoplay()
+    public bool IsActiveConversation(Conversation conversation)
     {
-        autoPlaying = true;
-        coroutine = StartCoroutine(AutoplayConversation());
+        if(conversation == null)
+        {
+            return false;
+        }
+
+        return activeConversations.ContainsKey(conversation);
```

Comment:
```suggestion
        return conversation != null && activeConversations.ContainsKey(conversation);
```
i think this will work? and be cleaner